p2p/session.go

```diff
@@ -259,7 +259,14 @@ func (s *session[H]) doRequest(
 }
 
 // processResponses converts HeaderResponse to Header.
-func (s *session[H]) processResponses(responses []*p2p_pb.HeaderResponse) ([]H, error) {
+func (s *session[H]) processResponses(responses []*p2p_pb.HeaderResponse) (h []H, err error) {
+	defer func() {
+		r := recover()
+		if r != nil {
+			err = fmt.Errorf("PANIC processing responses: %s", r)
```

Comment:
why not wrap with %w?

Reply:
Answering myself. Should not be a big deal if panic happened with string or error types.